src/Surface.zig

```diff
@@ -3870,22 +3870,35 @@ pub fn performBindingAction(self: *Surface, action: input.Binding.Action) !bool
         },
 
         .copy_to_clipboard => {
-            // We can read from the renderer state without holding
-            // the lock because only we will write to this field.
-            if (self.io.terminal.screen.selection) |sel| {
-                const buf = self.io.terminal.screen.selectionString(self.alloc, .{
-                    .sel = sel,
-                    .trim = self.config.clipboard_trim_trailing_spaces,
-                }) catch |err| {
-                    log.err("error reading selection string err={}", .{err});
-                    return true;
-                };
+            if (copyToClipboard(self)) {
+                return true;
+            }
+        },
+
+        .copy_or_interrupt => {
+            if (copyToClipboard(self)) {
+                return true;
+            } else {
+                const buf = try self.alloc.alloc(u8, 1);
```

Comment:
You can reduce the nesting level since the `then` branch of the `if` always returns.

Reply:
Fixed